frontend/src/metabase/app-main.js

```diff
@@ -24,6 +24,15 @@ init(reducers, getRoutes, store => {
     if (url.indexOf("/api/user/current") >= 0) {
       return;
     }
+
+    // If SSO is enabled, page url for login with email and password
+    // is `/auth/login/password` instead of `/auth/login`.
+    // So if call to api when signing in fails, let’s stay in the current page.
+    // Otherwise it will always redirect us to the Google auth interaction.
+    if (url === "/api/session") {
```

Comment:
let's use `url.includes("/api/session")` or similar like the above check on `/api/user/current` so that this doesn't fail on `/api/session/` (note the last char `/`)

Reply:
@daltojohnso, thanks.

We have a few variations of `/api/session/…` calls:

https://github.com/metabase/metabase/blob/a04d0cc21050bf66bf92f873bae586d5e6dc5855/frontend/src/metabase/services.js#L355-L363

I'd be concerned that checking for an `include` would be too encompassing. And without test coverage we might shoot our feet.

Updated with `_.contains` to check for either variation.
Wdyt?

Reply:
Also, watch for a potential security issue: https://codeql.github.com/codeql-query-help/javascript/js-incomplete-url-substring-sanitization/

Reply:
that works for me. since this piece of code is to prevent a redirect I think this code is fine, unless I am mistaken